dashbuilder-shared/dashbuilder-dataset-shared/src/main/java/org/dashbuilder/dataset/engine/function/MaxFunction.java

```diff
@@ -38,7 +38,10 @@ public Object aggregate(List values) {
             return null;
         }
 
-        List sorted = (List) values.stream().sorted().collect(Collectors.toList());
+        List sorted = (List) values.stream()
+                .filter(i -> i != null)
+                .sorted().collect(Collectors.toList());
```

Comment:
The difference still matters. Whereas max touches each element only once, sorting in general has to touch each element multiple times. I encourage you to explore the pefrormance impact of that for yourself using this [microbenchmark](https://gist.github.com/jhrcek/16308625c2e8a8a0b0267d367ab6d31d) for different values of N.